vsml_text_renderer/src/lib.rs

```diff
@@ -0,0 +1,218 @@
+        // 現状はデフォルト値を使用
+        let font_size = 32.0;
+        let line_height = 40.0;
+
+        let mut buffer = Buffer::new(&mut font_system, Metrics::new(font_size, line_height));
+
+        // フォントファミリーの設定
+        // TODO: フォールバックフォントは未対応
+        let font_family = if !style.font_family.is_empty() {
+            Family::Name(&style.font_family[0])
+        } else {
+            Family::SansSerif
+        };
+
+        let attrs = Attrs::new().family(font_family);
+
+        buffer.set_text(&mut font_system, text, &attrs, Shaping::Advanced);
+        buffer.shape_until_scroll(&mut font_system, false);
+
+        // 行の範囲とグリフの横幅を計算
+        let mut min_x = i32::MAX;
+        let mut max_x = i32::MIN;
+        let mut min_y = 0.0f32;
+        let mut max_y = 0.0f32;
+
+        for run in buffer.layout_runs() {
+            // 行の範囲を更新
+            min_y = min_y.min(run.line_top);
+            max_y = max_y.max(run.line_top + run.line_height);
+
+            for glyph in run.glyphs.iter() {
+                let physical_glyph = glyph.physical((0.0, run.line_y), 1.0);
+
+                if let Some(image) =
+                    swash_cache.get_image(&mut font_system, physical_glyph.cache_key)
+                {
+                    let glyph_x = physical_glyph.x + image.placement.left;
+
+                    min_x = min_x.min(glyph_x);
+                    max_x = max_x.max(glyph_x + image.placement.width as i32);
+                }
+            }
+        }
+
+        // バッファサイズを決定（行の高さ全体を使用）
+        let width = if max_x > min_x {
+            (max_x - min_x) as u32
+        } else {
+            1
+        };
+        let height = (max_y - min_y).ceil() as u32;
+        let offset_y = min_y as i32;
+
+        // RGBAバッファを作成（透明で初期化）
+        let mut rgba_buffer = vec![0u8; (width * height * 4) as usize];
+
+        // テキストの色を取得（デフォルトは白）
+        let text_color = style.color.unwrap_or((255, 255, 255, 255));
+
+        // cosmic-textでテキストをラスタライズ（2回目のイテレーション）
+        for run in buffer.layout_runs() {
+            for glyph in run.glyphs.iter() {
+                let physical_glyph = glyph.physical((0.0, run.line_y), 1.0);
+
+                if let Some(image) =
+                    swash_cache.get_image(&mut font_system, physical_glyph.cache_key)
+                {
+                    let glyph_x = physical_glyph.x + image.placement.left - min_x;
+                    let glyph_y = physical_glyph.y - image.placement.top - offset_y;
+
+                    // グリフの各ピクセルをRGBAバッファに描画
+                    for (pixel_y, row) in image
+                        .data
+                        .chunks(image.placement.width as usize)
+                        .enumerate()
+                    {
+                        for (pixel_x, &alpha) in row.iter().enumerate() {
+                            let x = pixel_x.checked_add_signed(glyph_x as isize).unwrap();
+                            let y = pixel_y.checked_add_signed(glyph_y as isize).unwrap();
```

Comment:
`glyph_x` `glyph_y`がマイナスになる想定はした方が良いかも
```suggestion
                            let Some(x) = pixel_x.checked_add_signed(glyph_x as isize) else { continue; };
                            let Some(y) = pixel_y.checked_add_signed(glyph_y as isize) else { continue; };
```